src/dram_controller.cc

```diff
@@ -552,21 +552,44 @@ bool MEMORY_CONTROLLER::add_wq(const request_type& packet)
   return false;
 }
 
+unsigned long DRAM_ADDRESS_MAPPING::swizzle_bits(champsim::address address, unsigned long segment_size, champsim::data::bits segment_offset, unsigned long field, unsigned long field_bits) const {
+  champsim::address_slice row{get<SLICER_ROW_IDX>(address_slicer), address};
+  unsigned long permute_field = field;
+
+  for(champsim::dynamic_extent subextent{champsim::data::bits{0}, segment_size}; subextent.upper <= row.upper_extent(); subextent = champsim::dynamic_extent{subextent.upper, segment_size}) {
```

Comment:
```suggestion
  using namespace champsim::data::data_literals;
  for(champsim::dynamic_extent subextent{0_b, segment_size}; subextent.upper <= row.upper_extent(); subextent = champsim::dynamic_extent{subextent.upper, segment_size}) {
```

By the way, I don't know if you know that this syntax exists. If you want it, here it is.